features/environment.py

```diff
@@ -40,17 +70,11 @@ def before_scenario(context, scenario):
     }
     context.starters_paths = starters_paths
     subprocess.run([context.pip, "install", "-r", "test_requirements.txt"])
-    context.temp_dir = Path(tempfile.mkdtemp())
+    context.temp_dir = Path(tempfile.mkdtemp()).resolve()
+    _PATHS_TO_REMOVE.add(context.temp_dir)
 
 
 def after_scenario(context, scenario):
-    rmtree(str(context.temp_dir))
-    rmtree(str(context.venv_dir))
-
-
-def rmtree(top):
-    if os.name != "posix":
-        for root, _, files in os.walk(top, topdown=False):
-            for name in files:
-                os.chmod(os.path.join(root, name), stat.S_IWUSR)
-    shutil.rmtree(top)
+    for path in _PATHS_TO_REMOVE:
+        # ignore errors when attempting to remove already removed directories
+        shutil.rmtree(path, ignore_errors=True)
```

Comment:
Is this just refactoring or necessary to make it work? `context` and `scenario` are now redundant arguments. Does it still belong to `after_scenario` or it should be `after_all`?

Reply:
With how behave is set up it always passes the context through, but `scenario` was indeed redundant and has been removed. Thanks for the catch!